Tests/SwiftParserTest/AttributeTests.swift

```diff
@@ -251,6 +251,89 @@ final class AttributeTests: XCTestCase {
     )
   }
 
+  func testIgnoreDeprecationAttribute() throws {
+    AssertParse("""
+    @available(*, deprecated)
+    class XClass {
+      var num: Int
+      init(_ num: Int) { self.num = num }
+    }
+    
+    @ignoreDeprecations(XClass)
+    func x() {
+      print(XClass(3))
+    }
+    """)
+
+    AssertParse("""
+    struct DeviceConf: Codable {
+
+      enum DeviceStatus: Codable {
+        case sandboxed
+        case partiallyEscaped
+        case fullyUnsandboxed
+      }
+
+      var isAMFIEnabled: Bool
+      var bootArgs: [String]
+
+      @available(*, deprecated)
+      var status: DeviceStatus
+
+      @ignoreDeprecation(status)
+      var canStartSession: Bool {
+        return status == .fullyUnsandboed
+      }
+    }
+    
+    """)
+
+    AssertParse("""
+    
+    @available(*, deprecated)
+    func deprecatedFunction() -> Int32 {
+      return 3
+    }
+    
+    @ignoreDeprecation(deprecatedFunction)
+    class StateManager {
+    
+      func someFunction() {
+        print(deprecatedFunction())
+      }
+    }
+    """)
+    
+    
+    AssertParse("""
+    enum GoatAlbums {
+      case afterHours
+      case theLifeOfPablo
+      case toPimpAButterfly
+
+      @available(*, deprecated)
+      case graduation
+    }
+    
+    @ignoreDeprecation(GoatAlbums.graduation)
+    func printBestAlbum() {
+      print(GoatAlbums.graduation)
+    }
```

Comment:
Sorry I should specified when I said enum element, I meant a test with the attribute on the case itself (I guess this could be used to silence a deprecation warning in a default value?). Really I'm just enumerating all the cases you've added as attribute options, i.e:

```
OnGenericType, OnEnumElement, OnImport, OnExtension, OnSubscript, OnVar, OnFunc,
```

IMO it would be nice to have a test for all of these.